docs/uiSchema.md

```diff
@@ -0,0 +1,182 @@
+# UiSchema
+
+A UI schema is basically an object literal following the tree structure of the form field hierarchy. It provides information on how the form should be rendered, while the JSON schema tells what.
+
+## **Custom widget**
+
+The uiSchema `ui:widget` property allows to define which widget should be rendered.
+
+Example:
+
+```js
+const schema = {
+  type: 'object',
+  properties: {
+    id: {
+      type: 'string',
+    }
+  }
+};
+
+const uiSchema = {
+  id: {
+    'ui:widget': 'IdWidget'
+  }
+};
+```
+
+To pass custom props to the widget you can use `ui:options` property.
+
+Example:
+
+```js
+const schema = {
+  type: 'object',
+  properties: {
+    geo: {
+      type: 'object',
+      items: {
+        properties: {
+          lat: {
+            type: 'number',
+          },
+          lon: {
+            type: 'number',
+          }
+        }
+      }
+    }
+  }
+};
+
+const uiSchema = {
+  geo: {
+    'ui:widget': 'MapWidget',
+    'ui:options': {
+      zoom: 10,
+    }
+  }
+};
+```
+
+**List of supported widgets:**
+
+- TextWidget
+- CheckboxWidget
+- SwitchWidget
+- SelectWidget
+- NumberWidget
+- RangeWidget
+- DateWidget
+- DateTimeWidget
+- IdWidget
+- MapWidget
+- YAMLWidget
+- hidden
+
+## **Show/hide individual fields**
+
+TODO:
+
+- [ ] add description
+- [ ] add examples of regular expressions:
+  - [ ] hide fields for specified values
+  - [ ] hide fields for values other than one specific
+  - [ ] hide fields when value is not selected
+
+```js
+const schema = {
+  type: object,
+  properties: {
+    account_type: {
+      type: 'string',
+      enum: [
+        'admin',
+        'user',
+      ]
+    },
+    email: {
+      type: 'string',
+    },
+    password: {
+      type: 'string',
+    },
+    username: {
+      type: 'string'
+    }
+  }
+}
+
+const uiSchema = {
+  account_type: {
+    'ui:logic': [
+      {
+        test: 'admin', // You can use regular expression here
+        hide: ['username']
+      }
+    ]
+  }
+}
+
+```
+
+## **Dynamic data fetching**
```

Comment:
In this point We need consider when data should be updated and where url should be defined. Also please define some option to set 'on demand' data in forms.

Reply:
What with related data to some field in forms?